traefik/templates/rbac/clusterrole.yaml

```diff
@@ -37,8 +37,8 @@ rules:
       - ""
     resources:
       - secrets
-    {{- if gt (len .Values.rbac.secretResourceNames) 0 }}
-    resourceNames: {{ .Values.rbac.secretResourceNames }}
+    {{- if .Values.rbac.secretResourceNames }}
+    resourceNames: {{ toYaml .Values.rbac.secretResourceNames |nindent 6 }}
```

Comment:
```suggestion
    {{- with .Values.rbac.secretResourceNames }}
    resourceNames: {{ toYaml . |nindent 6 }}
```

Reply:
the reason I didn't use `with` is because that adds a `resourceNames: []` stanza. I'm not sure how k8s interprets an empty array versus a nil array and leaving the if in place allows the tests to pass with no changes.

Reply:
ah yes. I believed default was `  secretResourceNames: `, but in fact, it's `  secretResourceNames: []`.
Thanks. :+1:

Reply:
@justinrush I tested it and it works using `with`.